core/state/state_object.go

```diff
@@ -207,6 +207,37 @@ func (s *stateObject) GetCommittedState(key common.Hash) common.Hash {
 	return value
 }
 
+// StateExists returns the flag indicating the storage slot is existent or not.
+func (s *stateObject) StateExists(key common.Hash) bool {
+	// Short circuit if the slot can be found in either dirty map (modified within
+	// the current transaction) or the pending storage (modified within the current
+	// block).
+	//
+	// Note, the slot is regarded as existent even they are marked as deleted,
+	// specifically the value is [32]byte{}, as there is no delete notion in verkle
+	// world anymore and the deletions will be translated to a special marker
+	// writing ultimately.
```

Comment:
If zeroes / deleted counts as existent, in verkle but not in mpt, then I think the high-level description of this operation (in the docstring) needs to be clarified. What are the semantics that we're supposed to deliver in this method?